File: MvpJava/src/main/java/com/droid/ui/BaseMvpActivity.java
package com.droid.ui;

import com.droid.presenter.BasePresenter;
import com.droid.presenter.BaseView;

public class BaseMvpActivity<T extends BasePresenter> extends BaseActivity implements BaseView {
    @Override
    public void showLoading() {

    }

    @Override
    public void hideLoading() {

    }

    @Override
    public void onError() {

    }

    public T mPresenter;
}

File: MvpKotlin/src/main/java/com/droid/mvp/ui/BaseMvpActivity.kt
package com.droid.mvp.ui

import com.droid.mvp.presenter.BasePresenter
import com.droid.mvp.presenter.BaseView

open class BaseMvpActivity<T:BasePresenter<*>>:BaseActivity(), BaseView {
    override fun showLoading() {

    }

    override fun hideLoading() {

    }

    override fun onError() {

    }

    lateinit var mPresenter:T
}